File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/get-game-result/get-game-result-response.ts
import type { BaseListResponse } from "~/interfaces/api/contracts/base/responses/base-response";
import type { GetGameResultResponseDto } from "./get-game-result-response-dto";

export class GetGameResultResponse implements BaseListResponse<GetGameResultResponseDto> {
    DataValues: GetGameResultResponseDto[];

    constructor(data?: GetGameResultResponseDto[]) {
        this.DataValues = data || [];
    }
}

File: HorseRacing.Contracts/Models/Game/Responses/GetGameResult/GetGameResultResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.GetGameResult
{
    public class GetGameResultResponse : BaseListResponse<GetGameResultResponseDto>
    {
        public GetGameResultResponse() : base() { }

        public GetGameResultResponse(List<GetGameResultResponseDto> data) : base(data) { }
    }
}
